test/test_bson_corpus.py

```diff
@@ -170,12 +186,19 @@ def run_test(self):
                         normalized_cE)
 
                 if 'lossy' not in valid_case:
-                    self.assertEqual(encode_bson(decode_extjson(E)), cB)
-
-                    if E != cE:
-                        self.assertEqual(
-                            encode_bson(decode_extjson(cE)),
-                            cB)
+                    # Skip tests for document type in Python 2.6 that have
+                    # multiple keys, since we can't control key ordering when
+                    # parsing JSON.
+                    if json_util._HAS_OBJECT_PAIRS_HOOK or not (
+                            sys.version_info[:2] == (2, 6) and
+                            bson_type == '0x03' and
+                            len(decode_extjson(E)) > 1):
+                        self.assertEqual(encode_bson(decode_extjson(E)), cB)
+
+                        if E != cE:
+                            self.assertEqual(
+                                encode_bson(decode_extjson(cE)),
+                                cB)
 
             for decode_error_case in case_spec.get('decodeErrors', []):
```

Comment:
This is overindented - it runs once per valid case or, if there are no valid cases, it doesn't run at all. That's why top.json seems to pass, even though the "Document with keys that start with $" test isn't right.